### PR TITLE
[Hub Generated] Review request for Microsoft.ContainerRegistry to add version preview/2023-06-01-preview

### PR DESCRIPTION
**Pull request by jikuma**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/jikuma-containerregistry-Microsoft.ContainerRegistry-2023-06-01-preview...main/).
### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version). Note that this doesn't apply if you are trying to merge a PR that was previously in the private repository.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/overview_breakingchanges).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](http://aka.ms/aprwiki))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by jikuma** [comment]
> Hi @jikuma, one or multiple validation error/warning suppression(s) is detected in your PR. Please follow the [Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) to get approval.

WI to fix the suppression [Task 24264387](https://msazure.visualstudio.com/AzureContainerRegistry/_workitems/edit/24264387): Use ARM common error response instead of ACR Error Response.

**Comment by keryul** [comment]
@raych1 please review the GO breaking change
@qiaozha please review the JS breaking change

**Comment by zizw123** [comment]
Approving the suppression and the PR based on our discussion and https://github.com/Azure/azure-rest-api-specs/pull/22777

**Comment by zizw123** [comment]
@jikuma The PR is good to go, if can you remove the suppression and we can approve the PR without the suppression, so that this can be tracked via S360

**Comment by jikuma** [comment]
Thanks you @zizw123, I have started the deployment with the new api version, once the ARM manifest is applied, we can merge the swagger.

**Comment by jikuma** [comment]
@rkmanda RP deployment with this change is complete. Can we merge this PR now. Please le me know if you need more info.

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2023-06-01-preview/containerregistry.json
Comment (kazrael2119) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2023-06-01-preview/containerregistry.json:4813]: You have two models with the same model name and here has a readOnly principalId the other one https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json#L1946 doesn't have. which will be a problem in SDK generation.  Can you provide a unique name for it ?

same issue in：
1.https://github.com/Azure/azure-rest-api-specs/pull/24426/files#diff-ec8941a9db152dfb094da578bfa7506d87e572746536a97bce808e26fe641711R4818
and 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json#L1950

2.https://github.com/Azure/azure-rest-api-specs/pull/24426/files#diff-ec8941a9db152dfb094da578bfa7506d87e572746536a97bce808e26fe641711R4889 
and 
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json#L2013

Comment (jikuma) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2023-06-01-preview/containerregistry.json:4813]: Thank you for your comment @kazrael2119. I get the issue now. We can not rename the property as this property is part of contract between ARM, Managed identity ACR.

From what I understand, the way to solve this is

1. Create a new api version for containerregistry_build,json and make this property readonly.
2. Revert the readonly change in containerregistry.json for api version 2023-06-01-preview

Let me know if their are other options. Ideally I want these property to be readonly if we are already rolling out new api version but do not want to create a new api version just for this change. On the server side we are already threating this property as read-only.

Comment (qiaozha) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2023-06-01-preview/containerregistry.json:4813]: Did you treat principalId as readonly at the service backend in the version 2019-06-01-preview ? If yes, we can change  the refer when generated the code to the readonly ones, and claim it as non-breaking, because the previous non-readonly won't work anyway. If no, we will have to revert the readonly change for now.

Comment (jikuma) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2023-06-01-preview/containerregistry.json:4813]: On the server side, we have always treated this as read-only. The issue lies in Swagger, where we neglected to set this property as read-only. We are now aiming to rectify this.

> If yes, we can change the refer when generated the code to the readonly ones

Is this something I need to do, somewhere? I already got this breaking change approved.

Comment (qiaozha) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2023-06-01-preview/containerregistry.json:4813]: @jikuma Thank you for the info, I think nothing need from your side.
@kazrael2119 Could you help on the directive in readme.typescipt.md ? Thanks

**Comment by qiaozha** [comment]
@tadelesh We are planning to use this directive to fix the AutoGenerated issue avoid breaking  https://github.com/Azure/azure-rest-api-specs/pull/24905/files#diff-ffa8a1b9f0b1e7035317d1a86e8aef6a504ca280349c5ff8c85ce5971d2f7f43R21-R45

**Comment by Alancere** [comment]
> @tadelesh We are planning to use this directive to fix the AutoGenerated issue avoid breaking https://github.com/Azure/azure-rest-api-specs/pull/24905/files#diff-ffa8a1b9f0b1e7035317d1a86e8aef6a504ca280349c5ff8c85ce5971d2f7f43R21-R45

add directive to reademe.go.md: https://github.com/Azure/azure-rest-api-specs/pull/24929

**Comment by jikuma** [comment]
Breaking changes for go sdk and javascript sdk has been handled. Please let me know if I need to any more changes to merge the PR.

**Comment by visingla-ms** [comment]
Approving as previous reviewers had only concern about Suppression Approval, which is handled now.

**Review by rkmanda — APPROVED** [review]

**Pull request merged**
